### PR TITLE
fix: DM pack review - messages, implementations, and tests

### PR DESCRIPTION
**Pull request by dmikalova**
Review pass on DM pack cards:
- Updated game messages for various cards
- Fixed card implementations
- Added/updated tests (HannibalsMark, reveal messages)
- Updated docs/card-messages.md and test/AGENTS.md

---

> [!NOTE]
> **Medium Risk**
> Touches core game-action messaging (`AllocateDamageAction`, `DiscardTopOfDeckAction`, bonus-icon resolution) and rewrites several card abilities, so regressions could affect logs and card resolution across many games. Added/updated tests reduce risk but changes still span both engine and card logic.
> 
> **Overview**
> Improves and standardizes game log output for several multi-step/conditional interactions, including **per-target allocated damage logging**, more descriptive `discardTopOfDeck` messages (listing the actual discarded cards or `nothing`), and resolving `power` bonus icons with a proper bonus-icon AbilityContext.
> 
> Refactors multiple `14-DM` cards to produce clearer, branch-accurate messaging (using `preferActionPromptMessage`, `then` narration, and dynamic `effectArgs`) and updates a few implementations (notably `Hannibal's Mark` house-change effect scoping, `Trash Heap` reveal/discard sequencing, and using `discardTopOfDeck` for deck discards).
> 
> Updates documentation (`card-messages.md`, `AGENTS.md`) with new messaging patterns and adds/extends tests covering `Hannibal's Mark` control/house edge cases and new message expectations for allocated damage, discard-top-of-deck, and reveal flows.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7da46e802a64750cd1db34d19e3cd33faa3b2c7d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: server/game/cards/14-DM/AuthorizedRequisitions.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ class AuthorizedRequisitions extends Card {
             },
             then: {
                 alwaysTriggers: true,
-                gameAction: ability.actions.draw()
+                gameAction: ability.actions.draw(),
+                message: '{0} uses {1} to draw a card'
```
Comment (strib) [inline server/game/cards/14-DM/AuthorizedRequisitions.js]: This seems unnecessary.  It changes the test output from:

```
2026-05-03 12:52:05 player1 plays Authorized Requisitions
2026-05-03 12:52:05 player1 gains an amber due to Authorized Requisitions's bonus icon
2026-05-03 12:52:05 player1 uses Authorized Requisitions
2026-05-03 12:52:05 player1 draws 1 card
```

to this:

```
2026-05-03 12:52:48 player1 plays Authorized Requisitions
2026-05-03 12:52:48 player1 gains an amber due to Authorized Requisitions's bonus icon
2026-05-03 12:52:48 player1 uses Authorized Requisitions
2026-05-03 12:52:48 player1 uses Authorized Requisitions to draw a card
2026-05-03 12:52:48 player1 draws 1 card
```

Do we need that extra line?

File: server/game/cards/14-DM/BrineReckoning.js [hunk 2]
```diff
@@ -4,18 +4,22 @@ class BrineReckoning extends Card {
...
-                target: [
-                    ...context.player.deck.slice(0, Math.min(context.player.deck.length, 5)),
-                    ...(context.player.opponent
+            preferActionPromptMessage: true,
+            gameAction: ability.actions.sequential([
+                ability.actions.discard((context) => ({
+                    location: 'deck',
+                    target: context.player.deck.slice(0, Math.min(context.player.deck.length, 5))
+                })),
+                ability.actions.discard((context) => ({
```
Comment (strib) [inline server/game/cards/14-DM/BrineReckoning.js]: Might be nice if this one specified it was from the opponent deck, right now it just says the player1 discard them, so it's a little confusing.

File: server/game/cards/14-DM/ColtSleven.js [hunk 3]
```diff
@@ -11,7 +11,15 @@ class ColtSleven extends Card {
                     (sum, card) => sum + card.powerCounters,
                     0
                 )
-            }))
+            })),
+            effect: 'deal 1 damage to a creature {1} time{2}',
```
Comment (strib) [inline server/game/cards/14-DM/ColtSleven.js]: Maybe we should have allocateDamage print out the actual damage to creatures.  (Can be a ticket to fix later.)

**Review by strib — APPROVED** [review]
